tests/unit_tests/test_time_series.py

```diff
@@ -98,7 +99,62 @@ def test_check_data_orientation_unbounded_maxshape(tmp_path):
         assert check_data_orientation(time_series) is None
 
 
-def test_check_timestamps():
+def test_check_timestamps_match_first_dimension_good():
+    assert (
+        check_timestamps_match_first_dimension(
+            time_series=pynwb.TimeSeries(
+                name="test_time_series",
+                unit="test_units",
+                data=np.empty(shape=4),
+                timestamps=[1.0, 2.0, 3.0, 4.0],
+            )
+        )
+        is None
+    )
+
+
+def test_check_timestamps_match_first_dimension_special_skip(tmp_path):
+    """
+    Very special skip condition for a certain older practice for indexing repeated Images.
+
+    The use of an ImageSeries for this is discouraged, with preference to use a stack of unordered Images instead.
+    """
+    nwbfile_path = tmp_path / "test_check_timestamps_match_first_dimension_special_skip.nwb"
+
+    nwbfile = make_minimal_nwbfile()
+    num_images = 5
+    image_width = 10
+    image_height = 15
+    num_channels = 3
+    dtype = "uint8"
+    image_series = pynwb.image.ImageSeries(
+        name="ImageSeries",
+        unit="n.a.",
+        data=np.empty(shape=(num_images, image_width, image_height, num_channels), dtype=dtype),
+        timestamps=[],
+    )
+    nwbfile.add_acquisition(image_series)
+    nwbfile.add_acquisition(
+        pynwb.image.IndexSeries(
+            name="IndexSeries",
+            unit="n.a.",
+            data=np.empty(shape=num_images),
```

Comment:
This generates floats, but the data should be `np.int32`. Also, I think `np.empty` is cool but I'm not sure it's the best thing to use here. In the future I'd like to build a check that the values of `data` are within the range of the length of the target, which would break this. Instead, we could just have it be `[0, 1]` have have the `timestamps` also have a length of 2. Having the timestamps depend on the number of images also doesn't make much sense here.

```suggestion
            data=[0,1],
```